frappe/desk/doctype/desktop_icon/desktop_icon.py

```diff
@@ -126,9 +126,10 @@ def add_user_icon(_doctype, _report=None, label=None, link=None, type='link', st
 		idx = frappe.db.sql('select max(idx) from `tabDesktop Icon` where owner=%s',
 			frappe.session.user)[0][0] or \
 			frappe.db.sql('select count(*) from `tabDesktop Icon` where standard=1')[0][0]
+		
+		doctype_icon = frappe.db.get_value('DocType', _doctype, ['icon','color','module'], as_dict=True)
```

Comment:
doctype icons are deprecated. this might change the view of a lot of existing icons. Not sure its a good idea. 

Maybe a patch to remove all existing icons?

Reply:
I can add a patch to remove existing icons. But lets do this else all the doctypes gets the same icon as module.

Reply:
@rmehta Added the patch to delete existing icon set in doctype.